### PR TITLE
Fix bug in `int()` and `float()` intrinsic functions

### PR DESCRIPTION
**Pull request by namannimmo10**

### CODE REVIEW

File: src/lpython/semantics/python_ast_to_asr.cpp [hunk 1]
```diff
@@ -3376,12 +3376,15 @@ class BodyVisitor : public CommonVisitor<BodyVisitor> {
             return (ASR::asr_t *)ASR::down_cast<ASR::expr_t>(ASR::make_Cast_t(
             al, loc, arg, ASR::cast_kindType::LogicalToInteger,
             to_type, value));
-        } else if (!ASRUtils::is_integer(*type)) {
+        } else if (ASRUtils::is_integer(*type)) {
+            return (ASR::asr_t *)arg;
```
Comment (czgdp1807) [inline src/lpython/semantics/python_ast_to_asr.cpp:3380]: So the kind will be retained here, right? If input is 32 bit integer then answer will also be 32 bit integer and if the input is 64 bit integer then answer will also be 64 bit integer.

**Comment by czgdp1807** [comment]
@namannimmo10 Do you want to continue working on this?

**Comment by namannimmo10** [comment]
Yes, I will complete this today!

File: src/lpython/semantics/python_ast_to_asr.cpp (hunk 1 above)
Comment (namannimmo10) [reply src/lpython/semantics/python_ast_to_asr.cpp:3465]: The kind would be retained for `int()`, whereas `float()` now gives only 64-bit numbers.

Comment (czgdp1807) [reply src/lpython/semantics/python_ast_to_asr.cpp:3465]: Why the difference in the behaviour of `int()` and `float()`? Is it not good to keep the behaviour for `int()` and `float()` similar?

**Comment by namannimmo10** [comment]
@czgdp1807 -- please feel free to merge if this looks good to you.

**Pull request merged**